e2etest fixes

altergui opened the pull request:
* apiclient: fix default CostExponent in NewElection
* remove unnecessary use of time.Sleep()
* elections can now be created with more than 2 choices 
* overwrite now votes different options, so now it's clear from results
 which vote was taken into account in case something unexpected happens
* avoid a panic in matchResults

mariajdab commented on cmd/end2endtest/overwrite.go:
```diff
@@ -75,19 +75,17 @@ func (t *E2EOverwriteElection) Run() error {
 	// overwrite the previous vote (choice 0) associated with account of index 0, using enough time to do it in the nextBlock
 	// try to make 3 overwrites (number of choices passed to the method). The last overwrite should fail due the maxVoteOverwrite constrain
-	err := t.overwriteVote([]int{0, 1, 0}, 0, nextBlock)
+	err := t.overwriteVote([]int{0, 1, 2}, 0, nextBlock)
```
In `newTestElectionDescription` we have declared only 2 possible choices allowed (`0` and `1`), so I think this value of `2` will not be taken into account

altergui replied:
that's so true! good catch <3

i refactored newTestElectionDescription to solve this

altergui committed:
apiclient: fix default CostExponent in NewElection

altergui committed:
e2etest: small fixes

* remove unnecessary use of time.Sleep()
* overwrite now votes different options, so now it's clear from results
  which vote was taken into account in case something unexpected happens
* avoid panic in matchResults, using reflect.DeepEqual instead,
  and refactor ballotVotes to remove hardcoded expectations about resultsFields
* elections can now be created with more than 2 choices

mariajdab commented on cmd/end2endtest/helpers.go:
```diff
@@ -602,15 +601,7 @@ func faucetPackage(faucet, faucetAuthToken, myAddress string) (*models.FaucetPac
-		}
-	}
-	return true
+	return reflect.DeepEqual(results, expectedResults)
```
Just to confirm, using reflect is not slower?

altergui replied:
i don't know. but in any case it's executed just once per test :shrug:
